charms/kubernetes_snaps.py

```diff
@@ -433,7 +468,103 @@ def configure_scheduler(extra_args_config, kubeconfig):
     with open(kube_scheduler_config_path, "w") as f:
         yaml.safe_dump(scheduler_config, f)
 
-    configure_kubernetes_service("kube-scheduler", scheduler_opts, extra_args_config)
+    config_files = {
+        scheduler_opts["config"],
+        kubeconfig,
+        kube_scheduler_config_path,
+    }
+
+    configure_kubernetes_service(
+        "kube-scheduler", scheduler_opts, extra_args_config, config_files
+    )
+
+
+@lru_cache
+def _sha256_file(config_file):
+    h = hashlib.sha256()
+    if Path(config_file).exists():
+        with open(config_file, "rb") as file:
+            while True:
+                # Reading is buffered, so we can read smaller chunks.
+                chunk = file.read(h.block_size)
+                if not chunk:
+                    break
+                h.update(chunk)
+    return h
```

Comment:
If the config path does not exist, a empty hash object is returned instead of an error. Is this desirable?

Reply:
Well, that's the idea.  The file that the kubernetes service depends on may or may not exist.  If it changes from existing to not existing -- that's an interesting change that should cause a difference...

but it looks like an empty file and a missing file return the same hash value -- so likely i should address that somehow...

Reply:
thanks to @eaudetcobello i've updated this a bit to provide a different hash when a file is empty from when it's missing.  it also will now include the file path as part of the hashed value as well as doing away with the `lrucache` in case files the files are changing multiple times during the reconcile loop